core/src/main/scala/org/apache/spark/storage/DiskBlockManager.scala

```diff
@@ -94,7 +98,16 @@ private[spark] class DiskBlockManager(
       } else {
         val newDir = new File(localDirs(dirId), "%02x".format(subDirId))
         if (!newDir.exists()) {
-          Files.createDirectory(newDir.toPath)
+          val path = newDir.toPath
+          Files.createDirectory(path)
+          if (shuffleServiceRemoveShuffleEnabled) {
+            // SPARK-37618: Create dir as group writable so files within can be deleted by the
+            // shuffle service in a secure setup. This will remove the setgid bit so files created
+            // within won't be created with the parent folder group.
+            val currentPerms = Files.getPosixFilePermissions(path)
+            currentPerms.add(PosixFilePermission.GROUP_WRITE)
+            Files.setPosixFilePermissions(path, currentPerms)
```

Comment:
I wonder why we need this for shuffle files but not RDD files previously? Or I miss it somewhere? cc @attilapiros

Reply:
If you're talking cache files served from the shuffle service, I expect they have the same issue and would just fail to delete when uncached in a secure environment.

Reply:
Yes, I meant the cache file, basically, it's generated by RDD.

@attilapiros What do you think about the removal issue?

Reply:
Confirmed that RDD removal via shuffle service does not work in a secure yarn setup (unrelated to this PR)